whypfs.go

```diff
@@ -526,14 +526,18 @@ func (p *Node) setupReprovider() error {
 	}
 
 	var err error
-	p.System, err = provider.New(p.Datastore,
+	if p.System, err = provider.New(p.Datastore,
 		provider.Online(p.Dht),
 		provider.ReproviderInterval(p.Config.ReprovideInterval),
 		provider.KeyProvider(provider.NewBlockstoreProvider(p.Blockstore)),
-	)
-	if err != nil {
+	); err != nil {
+		return err
+	}
+
+	if err := p.System.Reprovide(p.Ctx); err != nil {
```

Comment:
I really think you do not meant to do this.

`setupReprovider` [is called synchronously,](https://github.com/application-research/whypfs-core/blob/c9ab0a16770dbda6b5d29a350ebc2541113fce2a/whypfs.go#L257) `.Reprovide` synchronously waits for all data to be reprovided synchronously.

Reply:
@alvin-reyes `Reprovider.Run` is implicit when you use `Online` option.
I removed `.Run` because there was no world you would use `Online` but not `.Run`.

Reply:
You should be good if you just remove the call to reprovide and let `provider.New` do it's thing.